### DISCUSSION TITLE
Remove ability for tiers to never become undead

### DISCUSSION THREAD

**Discussion by srnyx** [commandgeek/geeksmp#44] [Ideas]
### What and How?

Make it so when a tier (subs, patron, etc...) is killed by an undead, the tier becomes undead.

### Why?

This will prevent players from farming kills and doing "free revives."

**What's a "free revive?"**
A free revive is when a tiered player lets an undead kill them so that the undead can become alive again.
There are no cons for either player as the tiered player stays alive even after being killed by the undead.

### Are there alternatives?

Completely removing the ability for tiered players to not become undead (if they die at all they become undead).
Kinda removes the point of buying a sub/becoming a Patron, so I think the suggested method above is a much better way.

### Checks

- [X] I have used the search at least once to check if my idea has already been suggested and perhaps already implemented.

**Comment by DWAA1660** [comment]
WHAT whats the point of buying subs and there is cons you lose all your levels and is still a pain

**Reply by srnyx** [reply]
Losing levels is not a big deal when:
1. Rarely need them (everyone already enchanted all of their stuff)
2. There are tons of enderman farms to use

**Comment by DDSH1231** [comment]
i agree with datnip..

**Comment by MousePlayerTwo** [comment]
Just, why. Players with tiers SHOULD have no cons, not becoming a undead is the only thing a tiered player gets from having that sort of rank...

**Reply by srnyx** [reply]
Tiered players wouldn't become undead from natural deaths or PVP, only when killed by an undead (which rarely ever happens cause everyone has netherite gear).
It's stupid for undeads to be able to skip the whole process of **trying** to kill someone.
*Moving my separate comment into a reply.*

**Comment by DWAA1660** [comment]
> > Just, why. Players with tiers SHOULD have no cons, not becoming a undead is the only thing a tiered player gets from having that sort of rank...
> 
> Tiered players wouldn't become undead from natural deaths or PVP, only when killed by an undead (which rarely ever happens cause everyone has netherite gear). It's stupid for undeads to be able to skip the whole process of **trying** to kill someone.

its impossible for undeads to kill people unless we let them

**Reply by srnyx** [reply]
That's exactly my point...
Undeads are trapped as undeads unless a tiered player **lets** the undead kill them

**Reply by MillkSheik** [reply]
But what would be the fun in it? a undead just enter the game and try hunting someone, but everyone is in netherite, it cant kill anyone and just leave the server.

**Reply by DWAA1660** [reply]
yea its not like you can get geared to kill them you cant do **anything** on the server unless your alive

**Comment by DWAA1660** [comment]
that will make so many people quit when you cant kill a netherite player with a fist and nobody wants to revive them

**Reply by srnyx** [reply]
Not sure how that would make people quit. I don't really understand what you're saying...

**Reply by FlyingDrag0n** [reply]
He means, if a ranked player would get undead. Less people would give out "free revives" because they don't want to die (just like undeads, they watch where they step). So less people would get revived, which means less people can likely 'play' on the server and that would resolve to quitting players.

**Reply by DWAA1660** [reply]
exactly ^^^^^

**Comment by Lianecx** [comment]
True, and people literally paid for it, they shouldnt have cons

**Reply by srnyx** [reply]
It isn't a con?
Normal players always become undead. Tiered players don't. That'd be how it works. Not sure how it's a con.

**Reply by DWAA1660** [reply]
becoming an undead is a con

**Reply by srnyx** [reply]
> becoming an undead is a con

Everyone else can become undead, so no, it isn't a con.

**Reply by DWAA1660** [reply]
i baught a sub SO i dont have to become undead

**Reply by srnyx** [reply]
> i baught a sub SO i dont have to become undead

You won't become undead if you aren't killed by an undead. Which rarely ever happens. When was the last time you were killed by an undead?

**Reply by DWAA1660** [reply]
prob a day or 2 ago

**Comment by FlyingDrag0n** [comment]
If ranked people would get tier undead.. That kinda breaks the point of buying a tier.. And it also means less people would continue buying it. I see why srnyx wants to get rid of "free revive" because there's less actions going on.. But he should think of smth else.

**Comment by srnyx** [comment]
**What about this system that was discussed via Discord:**
- Member/no rank: 1 life
- Tier 1/VIP: 2 lives
- Tier 2/VIP+: 3 lives
- Tier 3/MVP: 4 lives
- Contributor (if added in-game): 3 lives
- Gues: 4 lives

*Higher value overwrites (ex: both contributor + tier 3 means you'd get 4 lives)*

**We would use one of these systems for determining when lives reset:**
1. Every time a player becomes an undead their lives reset (example: I'm an alive Tier 2, I die 3 times, I become undead, I kill an alive player, I become alive, repeat [lives back at 3])
2. Lives reset every day (example: I'm an alive Tier 2, I die once today, the next day my lives would reset back to 3 and I'd have to die 3 times instead of 2 times to become undead)
3. Lives reset every week
4. Lives reset every month (this correlates to subscriptions because subs and Patron pay monthly, so this would make the most sense)

Lives would be displayed in action bar or as a bossbar (each time you die the bossbar goes down).
Bossbar might be best as it shows it more visually.

**Reply by DWAA1660** [reply]
if this was a thing i ould give my lives away then kill another ranked then they'd kill me then its infinate lives again

**Comment by Lianecx** [comment]
Additional gui just for listing the remaining lives? A command for that would be much better. Resetting lives every month is definitely too long for the current number of lives.

**Reply by srnyx** [reply]
Not sure exactly how a GUI would work. Action-bar or bossbar (with a command to toggle it, `/actionbar` or `/bossbar` respectively) + a `/lives` command would be best.
Also, please reply to the comment that is relevant instead of making an entirely new one.
